pkg/cmd/cli/cmd/expose.go

```diff
@@ -84,7 +84,8 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 		NamespaceParam(namespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, false, kcmdutil.GetFlagStringSlice(cmd, "filename")...).
 		ResourceTypeOrNameArgs(false, args...).
-		Flatten().
+		// TODO: Still need to enable exposing non-existent resources upstream.
+		Flatten().RequireObject(false).
```

Comment:
You wanted to enable exposing resources that don't exist. I am going to remove this since it requires upstream work too and I won't do it now.